### DISCUSSION TITLE
Extend session client expiry

### DISCUSSION THREAD

**Discussion by johnpaulcas** [WalletConnect/walletconnect-monorepo#2949] [Developer Support]
I'm looking at the documentation on v2 but can't find a setting to extend session expiry, is this possible? or if its possible to like extend the current session expiry or make it configurable on signClient, FYI, we are using the client SDK on our backend, as currently, no SDK is available for UnrealEngine so we used API to bridge between protocols. thanks!

**Comment by franfernandez20** [comment]
Hi, I was looking for the same info. I can't find any documentation about it.

 Some discoveries @johnpaulcas ??

**Comment by glitch-txs** [comment]
Hi! the session can be extended by calling `signClient.extend`. The extend is for 7 days https://docs.walletconnect.com/2.0/web/web3wallet/wallet-usage#extend-a-session

**Reply by johnpaulcas** [reply]
hi @glitch-txs, thanks!, so there is no way to like pass how long it can be extended? Also I notice everytime that we restart the server all the session topic of the users account that already got the session become invalid, do you have work around regards on that?
